contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -303,14 +303,17 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         // Process the receipts that were included in the teleporter message by paying the
         // fee for the messages are reward to the given relayers.
         uint256 length = teleporterMessage.receipts.length;
-        for (uint256 i = 0; i < length; ++i) {
+        for (uint256 i; i < length;) {
             TeleporterMessageReceipt memory receipt = teleporterMessage
                 .receipts[i];
             _markReceipt(
                 warpMessage.sourceChainID,
                 receipt.receivedMessageID,
                 receipt.relayerRewardAddress
             );
+            unchecked {
```

Comment:
can you add a comment about why we're using unchecked increment here, same for below loop. Thinking if the minimal gas optimization is worth the readability.

Reply:
Sure!

Reply:
As you say @minghinmatthewlam